### PR TITLE
refactor: adjust variable domain entities

### PR DESCRIPTION
**Pull request by houssain-barouni**
## Description

same as https://github.com/camunda/camunda/pull/24750

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

relates to https://github.com/camunda/camunda/issues/24911

### CODE REVIEW

File: db/rdbms/src/main/java/io/camunda/db/rdbms/read/service/VariableReader.java [hunk 1]
```diff
@@ -37,10 +40,14 @@ public VariableEntity findOne(final Long key) {
...
-    final var hits = variableMapper.search(filter);
+  public SearchResult search(final VariableQuery query) {
+    final var dbSort = convertSort(query.sort(), VariableSearchColumn.VAR_KEY);
+    final var dbQuery =
+        VariableDbQuery.of(
+            b -> b.filter(query.filter()).sort(dbSort).page(convertPaging(dbSort, query.page())));
+    LOG.trace("[RDBMS DB] Search for variables with filter {}", query);
+    final var totalHits = variableMapper.count(dbQuery);
+    final var hits = variableMapper.search(dbQuery);
     return new SearchResult(hits, totalHits.intValue());
```
Comment (stoerti) [inline db/rdbms/src/main/java/io/camunda/db/rdbms/read/service/VariableReader.java:51]: ❌   In other refactorings for the keyset pagination we also got rid of the local SearchResult class and used the SearchQueryResult directly (see ProcessInstanceReader). That one then also includes the sortValues.

File: qa/integration-tests/src/test/java/io/camunda/it/rdbms/db/variables/VariableIT.java [hunk 2]
```diff
@@ -159,14 +162,16 @@ public void shouldFindVariableWithFullFilter(final CamundaRdbmsTestApplication t
...
+  private void assertVariableDbModelEqualToEntity(
+      final VariableDbModel dbModel, final VariableEntity entity) {
     assertThat(entity)
         .usingRecursiveComparison()
         .ignoringFields("bpmnProcessId", "processDefinitionId")
         .isEqualTo(dbModel);
-    assertThat(entity.bpmnProcessId()).isEqualTo(dbModel.processDefinitionId());
+    assertThat(entity.processDefinitionId()).isEqualTo(dbModel.processDefinitionId());
   }
 }
```
Comment (stoerti) [inline qa/integration-tests/src/test/java/io/camunda/it/rdbms/db/variables/VariableIT.java:177]: 🔧  Could you also include a test with the searchAfter feature? (see ProcessInstanceIT#shouldFindProcessInstanceWithSearchAfter)

**Review by stoerti — CHANGES_REQUESTED** [review]
Nice that you directly refactored the variables to support keyset pagination 👍

File: qa/integration-tests/src/test/java/io/camunda/it/rdbms/db/variables/VariableIT.java (hunk 2 above)
Comment (houssain-barouni) [reply qa/integration-tests/src/test/java/io/camunda/it/rdbms/db/variables/VariableIT.java:215]: I added `VariableIT#shouldFindVariableWithSearchAfter`, but it is failing as it is returning the first page when I request the nextPage. Do you have an idea why?
Also I have noticed that, if I include `variableKey` in the sorting, I get `IndexOutOfBoundException` in https://github.com/camunda/camunda/blob/66fb7310900a460402f8d073503eb9a3583925e4/db/rdbms/src/main/java/io/camunda/db/rdbms/read/service/AbstractEntityReader.java#L105
we may need to add some validation on the searchAfter field to make sure it is consistent with the applied sorting

Comment (houssain-barouni) [reply qa/integration-tests/src/test/java/io/camunda/it/rdbms/db/variables/VariableIT.java:215]: I fixed it , I was missing `<include refid="io.camunda.db.rdbms.sql.Commons.keySetPageFilter"/>` in the MyBatis mapper

**Review by stoerti — APPROVED** [review]

**Pull request merged**